contracts/social-tips/src/contract.rs

```diff
@@ -0,0 +1,1346 @@
+            .range_raw(deps.storage, None, None, Order::Ascending)
+            .count();
+
+        // Ensure that the user don't have to many pending tips.
+        if user_pending_tips_count >= config.max_pending_tips as usize {
+            return Err(ContractError::ToManyPendingTipsForUser {
+                application,
+                handle,
+            });
+        }
+
+        let key = (application.clone(), handle.clone(), sender.clone());
+        let replaced = tips.may_load(deps.storage, key.clone())?;
+
+        tips.replace(
+            deps.storage,
+            key,
+            Some(&PendingTip {
+                sender,
+                amount: funds,
+                block_height: env.block.height,
+            }),
+            replaced.as_ref(),
+        )?;
+
+        let mut response = Response::new()
+            .add_attribute(ATTRIBUTE_ACTION, ACTION_SEND_TIPS)
+            .add_attribute(ATTRIBUTE_TIP_COLLECTED, "true")
+            .add_attribute(ATTRIBUTE_TIP_AMOUNT, serialized_coins);
+
+        // Send back the funds of the replaced tip.
+        if let Some(replaced_tip) = replaced {
+            response = response.add_message(BankMsg::Send {
+                amount: replaced_tip.amount,
+                to_address: replaced_tip.sender.to_string(),
+            });
+        }
+
+        Ok(response)
+    };
+}
+
+pub fn claim_tips(
+    deps: DepsMut<DesmosQuery>,
+    info: MessageInfo,
+) -> Result<Response<DesmosMsg>, ContractError> {
+    let mut coins = Vec::<Coin>::new();
+    let querier = ProfilesQuerier::new(deps.querier.deref());
+    let pending_tips_map = pending_tips();
+
+    for app_link_result in
+        querier.iterate_application_links(Some(info.sender.clone()), None, None, 10)
+    {
+        let app_link = app_link_result?;
+        if app_link.state == ApplicationLinkState::VerificationSuccess {
+            let mut pending_tips = pending_tips_map
+                .prefix((
+                    app_link.data.application.clone(),
+                    app_link.data.username.clone(),
+                ))
+                .range(deps.storage, None, None, Order::Ascending)
+                .collect::<StdResult<Vec<_>>>()?;
+
+            for (sender, mut pending_tip) in pending_tips.drain(0..) {
+                pending_tip
+                    .amount
+                    .drain(0..)
+                    .for_each(|coin| coins.push(coin));
+                pending_tips_map.remove(
+                    deps.storage,
+                    (
+                        app_link.data.application.clone(),
+                        app_link.data.username.clone(),
+                        sender,
+                    ),
+                )?;
+            }
+        }
+    }
+
+    let merged_coins = sum_coins_sorted(coins)?;
+
+    if merged_coins.is_empty() {
+        return Err(ContractError::NoTipsAvailable {
+            user: info.sender.to_string(),
+        });
+    }
```

Comment:
It can be moved before `sum_coins_sorted` to return error without summing.

Reply:
as above